### ISSUE TITLE
[Phase 4] Toxicity Classification (Tox21 Benchmark)

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/molecular-property-prediction#5]
## Objective

Train multi-task toxicity classifier on Tox21 benchmark.

## Dependencies

- Phase 2: Descriptor Generation (requires molecular features)
- Phase 3: Solubility Prediction (can run in parallel)

## Tasks

- [ ] Implement `src/models/classifier.py` for toxicity classification
- [ ] Implement `src/evaluation/classification_metrics.py` for AUC-ROC
- [ ] Handle class imbalance (few positives in Tox21)
- [ ] Handle missing labels in some endpoints
- [ ] Train multi-task model for 12 endpoints
- [ ] Save trained model to `models/tox21_model_v1.joblib`
- [ ] Create `notebooks/03_toxicity_analysis.ipynb`

## Files to Create

| File | Purpose |
|------|---------|
| `src/models/classifier.py` | Toxicity classifier |
| `src/evaluation/classification_metrics.py` | Classification metrics |
| `notebooks/03_toxicity_analysis.ipynb` | Analysis |

## Target Metrics

| Metric | Target |
|--------|--------|
| Mean AUC | > 0.70 |
| Per-endpoint AUC | > 0.65 |

## Starter Code

```python
# src/models/classifier.py
"""Molecular toxicity classification models."""

import numpy as np
import joblib
from sklearn.ensemble import RandomForestClassifier
from sklearn.multioutput import MultiOutputClassifier

class ToxicityClassifier:
    """Multi-task classifier for toxicity endpoints."""

    def __init__(self, n_estimators: int = 100):
        self.base_model = RandomForestClassifier(
            n_estimators=n_estimators,
            class_weight="balanced",
            random_state=42
        )
        self.model = MultiOutputClassifier(self.base_model)

    def fit(self, X: np.ndarray, y: np.ndarray) -> "ToxicityClassifier":
        """Train on multiple endpoints."""
        # Handle NaN labels by masking
        self.model.fit(X, y)
        return self

    def predict(self, X: np.ndarray) -> np.ndarray:
        return self.model.predict(X)

    def predict_proba(self, X: np.ndarray) -> list[np.ndarray]:
        """Return probabilities for each endpoint."""
        return self.model.predict_proba(X)

    def save(self, path: str) -> None:
        joblib.dump(self.model, path)
```

## Definition of Done

- [ ] Mean AUC > 0.70 across endpoints
- [ ] Class imbalance handled appropriately
- [ ] Per-endpoint performance reported
- [ ] Comparison with MoleculeNet baselines
- [ ] All tests passing

## Technical Notes

- Tox21 has 12 toxicity endpoints
- Severe class imbalance (few positives)
- Some endpoints have missing labels - need masking strategy

---
Part of #1 (Meta Issue)
